app/model/MessageConstants.scala

```diff
@@ -94,7 +96,17 @@ object MessageConstants {
     val confirmSelectedAddressError2 = "search for and select your address"
 
     val lookupErrorHeading = "There is a problem"
-    val lookupPostcodeError = "The postcode you entered appears to be incomplete or invalid. Please check and try again."
+    val lookupPostcodeEmptyError: Boolean => String = isUkMode => if(isUkMode) "Enter UK postcode." else "Enter postcode."
+    val lookupPostcodeInvalidError: Boolean => String = isUkMode => if(isUkMode) {
+      "UK Postcode must only contains letters A - Z, and 0 - 9."
+    } else {
+      "Postcode must only contains letters A - Z, and 0 - 9."
+    }
+    val lookupPostcodeError: Boolean => String = isUkMode => if(isUkMode) {
+      "Enter a real UK Postcode e.g. AA1 1AA."
+    } else {
+      "Enter a real Postcode e.g. AA1 1AA."
+    }
```

Comment:
This was the simplest (or less intrusive) way I could find to get the job done done based on the prior art of  `val errorMax: Int => String = max => s"Maximum length is $max"`

However, it feels pretty gnarly and i'm convinced there must be a nicer way to handle a simple 3 character difference in the strings (for English at least) with something more along the lines of `def messageConstants(isWelsh:Boolean): MessageConstants = if(isWelsh) WelshMessageConstants else EnglishMessageConstants`

But again, I don't know enough scala to know what that would be without it becoming super complex. I have some ideas if this were JS but I'm not sure how well it would translate.

Reply:
Yeah, isUkMode being repeated for each method is not ideal. We can refactor the whole messages to take isUkMode flag and then we don't need to have it for each method. But its a little bigger refactor so might have to do it later as a separate jira.

Reply:
cool that's what i figured.

there are going to be more messages coming that will depend on ukMode so maybe we get them all in first and then refactor?